raiden/network/transport/matrix/utils.py

```diff
@@ -503,9 +505,11 @@ def validate_userid_signature(user: User) -> Optional[Address]:
 
     try:
         displayname = user.get_display_name()
-        recovered = recover(
-            data=user.user_id.encode(), signature=Signature(decode_hex(displayname))
-        )
+        if DISPLAY_NAME_HEX_RE.match(displayname):
```

Comment:
Is this new if check covered by any unit test?

Reply:
Yes, invalid signatures are already being tested for. This is just an optimization to skip ecrecover on obviously invalid values.